### PR TITLE
Switch to using pytest-variables

### PR DESCRIPTION
**Pull request by davehunt**
This probably breaks some Paypal login stuff, but as far as I could tell that's not being used. I'd prefer to remove this dead code or fix it up (and use it) in a separate patch.

### CODE REVIEW

**Comment by davehunt** [comment]
Dev desktop: http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/marketplace.variables.adhoc/2/
Dev mobile: http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/marketplace.mobile/10/

Results seem comparable to latest results without this patch. I'm more than happy waiting for outstanding issues affecting test results to be addressed first, but we should avoid too much change before landing this patch as it touches a lot of files.

File: conftest.py [hunk 1]
```diff
@@ -1,39 +1,54 @@
...
-def free_app(request):
-    """Return a free app created via the Marketplace API, and automatically delete the app after the test."""
-    mozwebqa = request.getfuncargvalue('mozwebqa')
-    request.app = MockApplication()
-    api = MarketplaceAPI.get_client(mozwebqa.base_url,
-                                    mozwebqa.credentials)
-    api.submit_app(request.app)
+@pytest.fixture
+def free_app(request, api):
+    """Return a free app created via the Marketplace API."""
```
Comment (bobsilverberg) [inline conftest.py]: Is there a reason you chose to remove the second part of the docstring? The part about deleting the app after the test?

File: tests/desktop/consumer_pages/test_users_account.py [hunk 2]
```diff
@@ -15,15 +15,11 @@
 
 class TestAccounts(BaseTest):
 
-    @pytest.mark.credentials
-    def test_create_new_user_using_API(self, mozwebqa):
-
+    def test_create_new_user_using_fxapom(self, mozwebqa, new_user):
```
Comment (bobsilverberg) [inline tests/desktop/consumer_pages/test_users_account.py]: The need for this test is questionable as we are exercising this same set of steps in a number of other tests (where we are creating a new FxA account and then using it to log in to Marketplace).

@krupa can we delete this test?

**Comment by bobsilverberg** [comment]
LGTM @davehunt. Nice work! When I run the full suite I do get a few failures, but none of them seem to be related to this PR and I know there are still some issue with the tests.

r+

**Comment by krupa** [comment]
Let us keep the test for now since it is testing a very distinct user action: Signing in. We can revisit this if people feel strongly about it.

File: conftest.py (hunk 1 above)
Comment (davehunt) [inline conftest.py]: Not a good one - I was fixing local line length warnings, and this seemed redundant information at the time. I can see why we'd want to include it though. I'll restore it.

**Comment by davehunt** [comment]
Merged in https://github.com/mozilla/marketplace-tests/commit/ef6320849a1f8bdfc04bf4adfb1c36b662527f5e

**Comment by bobsilverberg** [comment]
These changes have been merged, but this PR is still open, so I'm closing it.

**Pull request closed without merging**